include/oneapi/dpl/experimental/kt/two_pass_scan.h

```diff
@@ -230,68 +329,18 @@ two_pass_scan(sycl::queue q, _InRng&& __in_rng, _OutRng&& __out_rng,
-                        }
-                    }
-                    else if (is_full_thread)
-                    {
-                        auto v = unary_op(__in_rng[start_idx]);
-                        sub_group_scan<VL, IsInclusive, /*InitPresent=*/false>(sub_group, v, binary_op, sub_group_carry);
-
-                        for (int j = 1; j < J; j++)
-                        {
-                            v = unary_op(__in_rng[start_idx + j * VL]);
-                            // In principle we could use SYCL group scan. Stick to our own for now for full control of implementation.
-                            sub_group_scan<VL, IsInclusive, /*InitPresent=*/true>(sub_group, v, binary_op, sub_group_carry);
-                        }
-                    }
-                    else
-                    {
-                        auto iters = oneapi::dpl::__internal::__dpl_ceiling_div(M - subgroup_start_idx, VL);
-                        if (iters == 1)
-                        {
-                            auto local_idx = (start_idx < M) ? start_idx : M - 1; // use a dummy value for unused elements out of range
-                            auto v = unary_op(__in_rng[local_idx]);
-                            sub_group_scan_partial<VL, IsInclusive, /*InitPresent=*/false>(sub_group, v, binary_op, sub_group_carry,  M - subgroup_start_idx);
-                        }
-                        else
-                        {
-                            auto v = unary_op(__in_rng[start_idx]);
-                            // In principle we could use SYCL group scan. Stick to our own for now for full control of implementation.
-                            sub_group_scan<VL, IsInclusive, /*InitPresent=*/false>(sub_group, v, binary_op, sub_group_carry);
-
-                            for (int j = 1; j < iters - 1; j++)
-                            {
-                                v = unary_op(__in_rng[start_idx + j * VL]);
-                                sub_group_scan<VL, IsInclusive, /*InitPresent=*/true>(sub_group, v, binary_op, sub_group_carry);
-                            }
-
-                            auto local_idx = (start_idx + (iters - 1) * VL < M) ? start_idx + (iters - 1) * VL : M - 1;
-                            v = unary_op(__in_rng[local_idx]);
-
-                            sub_group_scan_partial<VL, IsInclusive, /*InitPresent=*/true>(sub_group, v, binary_op, sub_group_carry,
-                                                                        M - (subgroup_start_idx + (iters - 1) * VL));
-                        }
-                    }
-
+                    scan_through_elements_helper<VL, IsInclusive,
+                                                 /*InitPresent=*/false,
+                                                 /*CaptureOutput=*/false, J_max>(
+                        sub_group, unary_op, binary_op, sub_group_carry, __in_rng, nullptr, start_idx, M, J,
```

Comment:
I removed the `require` for `__out_rng` for this kernel, because it is unused in this kernel.  We would need to re-introduce that to (not) use it here.  I thought about just using `__in_rng`, but it seems that some empty type is better.  

Also, this `nullptr` attempts to make it clear that we will not be using this parameter or capturing anything to the output.

Would you prefer `std::void_t` perhaps?

Reply:
Ah, I forgot it's unused in this kernel. I don't think it is worth it to bring `__out_rng` in here then. I am fine with leaving it with `nullptr`. It is clear what that means to me.